### ISSUE TITLE
finish release error (Mac OS)

### ISSUE THREAD

**Issue by MarcoStagno** [OpherV/gitflow4idea#273]
* **I'm submitting a ...**
  - [X] bug report
  - [ ] feature request
  - [ ] puppy => You're not submitting a puppy. I already have one and he's [adorable](https://twitter.com/Opherv/status/974341249099542528)

* **What is the current behavior?**
Finish release doesn't work

* **Is this a bug? Sorry about that. If so give me explicit details how to reproduce:**
Try to finish a `release`: I got this error from console:
`12:16:20.224: [locapp] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false flow release finish -m "Tagging version 1.52.0" 1.52.0`
**Fatal: Branch 'release/version' does not exist and is required.**

* **What is the expected behavior?**
it'd finish the release

* **What is the motivation / use case for changing the behavior?**
I found that from the cli it works if I put the tag version before the -m
`git -c credential.helper= -c core.quotepath=false -c log.showSignature=false flow release finish 1.52.0 -m "Tagging version 1.52.0"` works fine

* **Please tell me about your environment:**
  
  - Gitflow4idea version: *find in IntelliJ settings plugins*
0.7.1 
  - Gitflow version: run in terminal `>git flow version`
1.12.3 (AVH Edition)
  
  - *IntelliJ Help -> about > click copy icon and paste here. Should look like this:*

~~~~
PhpStorm 2019.3.2
Build #PS-193.6015.48, built on January 22, 2020
Licensed to Sintesia S.r.l. / Marco Stagno
Subscription is active until March 16, 2020
Runtime version: 11.0.5+10-b520.30 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.15.2
GC: ParNew, ConcurrentMarkSweep
Memory: 1981M
Cores: 8
Registry: run.processes.with.pty=TRUE, ide.tooltip.initialDelay=999
Non-Bundled Plugins: Gitflow, org.psliwa.idea.composer, org.sylfra.idea.plugins.linessorter
~~~~
  

* **Other information** (e.g. detailed explanation, stacktrace, related issues, suggestions how to fix, links for me to have context words of praises, pictures of puppies (again with the puppy??) )

**Comment by OpherV** [comment]
@MarcoStagno  Can you paste the contents of your `config` file inside your `.git` folder, and also run `git branch -a` and paste its contents?

**Comment by MarcoStagno** [comment]
this is the `config`

```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = git@github.com:KanbanBOX/KanbanBOX.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[gitflow "branch"]
	master = master
	develop = development
[gitflow "prefix"]
	feature = feature/
	release = release/
	hotfix = hotfix/
	support = support/
	versiontag = 
	bugfix = bugfix/
[gitflow "path"]
	hooks = /Users/mas/www/locapp.kanbanbox.com/.git/hooks
[gitflow "branch.feature/fix_webhook_ftp"]
	base = development
[gitflow "branch.feature/fix_excel_template_lang"]
	base = development
[gitflow "branch.feature/fix_customer_backup_for_public_directory"]
	base = development
[gitflow "branch.feature/environment_in_mail_errori_s3"]
	base = development
[gitflow "branch.feature/better_escaping"]
	base = development
[gitflow "branch.feature/api_fix_negative_limit"]
	base = development
[gitflow "branch.feature/limit_graphs_dates_range"]
	base = development
[gitflow "branch.feature/new_secret_file"]
	base = development
[gitflow "branch.hotfix/better_production_db_config"]
	base = master
[gitflow "branch.feature/customer_backup_secrets_fix"]
	base = development
[gitflow "branch.feature/deployer"]
	base = development
[gitflow "branch.feature/fix_for_testlocapp"]
	base = development
[gitflow "branch.hotfix/fix_template_excel_consumi_lingue_straniere"]
	base = master
[gitflow "branch.feature/regole_di_stampa_su_template_cancellati"]
	base = development
[gitflow "branch.feature/faster_count_db_records"]
	base = development
[gitflow "branch.feature/webhook_on_filters"]
	base = development
[gitflow "branch.feature/motivo_di_blocco_su_email_notifica"]
	base = development
[gitflow "branch.hotfix/fix_filtri_webhook"]
	base = master
[gitflow "branch.feature/webhook_con_link_per_test"]
	base = development
[gitflow "branch.feature/fix_default_maintenace_widget"]
	base = development
[gitflow "branch.feature/fix_cron_rule_example_in_english_language"]
	base = development
[gitflow "branch.feature/webhook_id_in_log_table"]
	base = development
[branch "development"]
	remote = origin
	merge = refs/heads/development
[gitflow "branch.feature/new_upload_limit_for_file_manager"]
	base = development
[gitflow "branch.feature/export_excel_for_all_temp_import_tables"]
	base = development
[gitflow "branch.feature/new_upload_limit"]
	base = development
[gitflow "branch.feature/no_secret_data_in_logs"]
	base = development
[gitflow "branch.feature/fix_ftp_nlist_amibuity"]
	base = development
[gitflow "branch.feature/s3_fix_multiple_call_error"]
	base = development
[gitflow "branch.feature/codeception"]
	base = development
[gitflow "branch.feature/change_kb_license_owner"]
	base = development
[gitflow "branch.feature/switch_kb_owner_license"]
	base = development
[gitflow "branch.hotfix/add_more_urls_to_csrf_exclude_uris_list"]
	base = master
[gitflow "branch.feature/webhook_post_file"]
	base = development
[gitflow "branch.feature/fix_wrong_excel_type"]
	base = development
[gitflow "branch.feature/add_comment_warning"]
	base = development
[gitflow "branch.feature/fix_missing_lang"]
	base = development
[gitflow "branch.feature/fix_batch_release_event_with_add_extra_cards"]
	base = development
[gitflow "branch.feature/missing_langs"]
	base = development
[gitflow "branch.feature/add_dynamic_location_to_cards"]
	base = development
[gitflow "branch.feature/italian_language_in_all_graphs_excel_data"]
	base = development
[gitflow "branch.feature/fix_hardcoded_lang_in_api_post_custom_scripts"]
	base = development
[gitflow "branch.feature/composer"]
	base = development
[branch "feature/composer"]
	remote = origin
	merge = refs/heads/feature/composer
[gitflow "branch.feature/webhook_manual_trigger"]
	base = development
[gitflow "branch.feature/add_user_access_to_test_environment_only"]
	base = development
[gitflow "branch.feature/fix_wrong_model_place"]
	base = development
[gitflow "branch.feature/test"]
	base = development
[gitflow "branch.release/1.50.0"]
	base = development
[gitflow "branch.feature/no_widget_for_ftp_trigger_actions"]
	base = development
[gitflow "branch.feature/fix_time_zone_on_consumption_model"]
	base = development
[gitflow "branch.feature/fix_wrong_pop_on_wall"]
	base = development
[gitflow "branch.feature/fix_nesting_lang"]
	base = development
[gitflow "branch.feature/fix_mobile_captcha"]
	base = development
[gitflow "branch.feature/fix_email_validation_linking_partner"]
	base = development
[gitflow "branch.feature/add_email_single_data-type_to_custom_form"]
	base = development
[branch "feature/add_email_single_data-type_to_custom_form"]
	remote = origin
	merge = refs/heads/feature/add_email_single_data-type_to_custom_form
[branch "feature/fix_email_validation_linking_partner"]
	remote = origin
	merge = refs/heads/feature/fix_email_validation_linking_partner
[gitflow "branch.feature/new_help_page"]
	base = development
[gitflow "branch.feature/fix_erase_template_errors"]
	base = development
[gitflow "branch.feature/add_kanban_status_as_default_for_prod"]
	base = development
[gitflow "branch.feature/fix_default_widget_function"]
	base = development
[gitflow "branch.feature/api_new_worklist_cards_resource"]
	base = development
[branch "feature/api_new_worklist_cards_resource"]
	remote = origin
	merge = refs/heads/feature/api_new_worklist_cards_resource
[gitflow "branch.feature/webhook_new_trigger_date_field"]
	base = development
[gitflow "branch.feature/import_excel_data_in_a_better_wat"]
	base = development
[branch "feature/import_excel_data_in_a_better_wat"]
	remote = origin
	merge = refs/heads/feature/import_excel_data_in_a_better_wat
[gitflow "branch.feature/from_sintesia_to_kanbanbox"]
	base = development
[gitflow "branch.feature/fix_users_form_on_user_access"]
	base = development
[branch "feature/from_sintesia_to_kanbanbox"]
	remote = origin
	merge = refs/heads/feature/from_sintesia_to_kanbanbox
[gitflow "branch.feature/fix_template_undo"]
	base = development
[gitflow "branch.feature/webhook_worklist_on_combined_data"]
	base = development
[gitflow "branch.feature/fix_develompent"]
	base = development
[gitflow "branch.feature/fix_multiplant_add_user"]
	base = development
[gitflow "branch.feature/fix_generation_WL"]
	base = development
[gitflow "branch.feature/webhook_manager_fields"]
	base = development
```

and the result of `git branch -a`

```
  development
  feature/add_email_single_data-type_to_custom_form
  feature/api_new_worklist_cards_resource
  feature/codeception
  feature/composer
  feature/fix_email_validation_linking_partner
  feature/fix_template_undo
  feature/from_sintesia_to_kanbanbox
  feature/import_excel_data_in_a_better_wat
  feature/mysql8_test
* feature/webhook_manager_fields
  master
  remotes/origin/HEAD -> origin/master
  remotes/origin/development
  remotes/origin/feature/Penetration_Test
  remotes/origin/feature/add_email_single_data-type_to_custom_form
  remotes/origin/feature/api_new_worklist_cards_resource
  remotes/origin/feature/avg_consumption
  remotes/origin/feature/better_board_filtered
  remotes/origin/feature/fix_consumption_graph_timezone
  remotes/origin/feature/fix_develompent
  remotes/origin/feature/fix_email_validation_linking_partner
  remotes/origin/feature/from_sintesia_to_kanbanbox
  remotes/origin/feature/import_excel_data_in_a_better_wat
  remotes/origin/feature/invoices_fro_2_companies
  remotes/origin/feature/json_beautifier
  remotes/origin/feature/lock_documents_add_large_worklist_error
  remotes/origin/feature/negotiation_total_price_ajax_calculation
  remotes/origin/feature/new_flutter_release
  remotes/origin/feature/template_in_inches
  remotes/origin/master

```

**Comment by MarcoStagno** [comment]
the problem is also in 'finish hotfix'
`git -c credential.helper= -c core.quotepath=false -c log.showSignature=false flow hotfix finish -m "Fix  js increment number" 1.57.0`
`Fatal: Branch 'hotfix/js' does not exist and is required.`

**Comment by OpherV** [comment]
@MarcoStagno just to make sure I understand - in the first case where you tried to finish a feature, what branch were you on?

In the second case, when you tried to finish the hotfix, what branch were you on?

Also - the next time this happens, can you try and run the corresponding command from the console and see if you get the same error?
IE if finish hotfix fails, run something like
`git flow hotfix finish -m "my commit message" 1.0.0` and see what output you get  

I'm trying to ascertain whether the issue is endemic to the plugin, or something specific with your gitflow/project/system etc

**Comment by MarcoStagno** [comment]
@OpherV in both cases I was in the branch that I was trying to finish; in the first case I was on branch `release/1.52.0` and in the second case on `hotfix/1.57.0` 

I already run the same command on console and the error was the same; I had to change it manually (on console) to makes it works 

for example from:
`git flow hotfix finish -m "my commit message" 1.0.0`
to
`git flow hotfix finish  1.0.0 -m "my commit message" `

thank you for your support!

**Comment by OpherV** [comment]
Oh that is interesting
On the [AVH wiki](https://github.com/petervanderdoes/gitflow-avh/wiki/Reference:-git-flow-hotfix#git-flow-hotfix-finish---finish-a-hotfix-branch) it says that the version param is last, so seems there might be a bug there?

![image](https://user-images.githubusercontent.com/3951311/74567331-36acf680-4f7e-11ea-9b55-10f92c820648.png)


Anyhow I guess I can work around it by flipping the order of the parms. I'll have a look

**Comment by vincentvialard** [comment]
Hi there,

same problem same solution for me.

Here is the trace in the console:
```$ git -c credential.helper= -c core.quotepath=false -c log.showSignature=false flow release start v3.11.1
Switched to a new branch 'release/v3.11.1'
$ git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- parent/pom.xml
$ git -c credential.helper= -c core.quotepath=false -c log.showSignature=false commit -F /private/var/folders/zs/50_tqfb15xl4m7wlf9dmr0d40000gn/T/git-commit-msg-.txt --
[release/v3.11.1 d073814] release v3.11.1
 1 file changed, 1 insertion(+), 1 deletion(-)
$ git -c credential.helper= -c core.quotepath=false -c log.showSignature=false flow release finish -m "Tagging version v3.11.1" v3.11.1
Fatal: Branch 'release/version' does not exist and is required.
```

It might come from the git command parser...

The solution was to place the -m "message" after the version number:
```
$ git -c core.quotepath=false -c log.showSignature=false flow release finish v3.11.1 -m "Tagging version v3.11.1"
```
So we have a workaround...

Thanks for the plugin!

**Comment by OpherV** [comment]
Fixed in 0.7.2
